core/capabilities/remote/executable/parallel_execution_limiter.go

```diff
@@ -0,0 +1,53 @@
+package executable
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"golang.org/x/sync/semaphore"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+var ErrMaxParallelExecutionLimitReached = errors.New("parallel execution limit reached")
+
+type parallelExecutionLimiter struct {
+	maxParallelTasks int64
+	sem              *semaphore.Weighted
+	wg               sync.WaitGroup
+	stopChan         services.StopChan
+}
+
+func newParallelExecutionLimiter(maxParallelTasks int64) *parallelExecutionLimiter {
+	return &parallelExecutionLimiter{
+		maxParallelTasks: maxParallelTasks,
+		sem:              semaphore.NewWeighted(maxParallelTasks),
+		stopChan:         make(services.StopChan),
+	}
+}
+
+// ExecuteTask executes a task in parallel, returning ErrMaxParallelExecutionLimitReached if the maximum number of
+// parallel executing tasks has been reached.  The task is executed with the given context.
+func (t *parallelExecutionLimiter) ExecuteTask(ctx context.Context, task func(ctx context.Context)) error {
```

Comment:
Where are we calling this?

Reply:
@ettec I think Bolek is asking where is ExecuteTask being called from? (I search for ExecuteTask myself and don't see where we call it)